tools/harvest_source_import/remote_ckan/lib.py

```diff
@@ -0,0 +1,217 @@
+
+        for hs in harvest_sources:
+            title = hs['title']
+            source_type = hs['source_type']  # datajosn, waf, etc
+            state = hs['state']
+            name = hs['name']
+            
+            logger.info(f'  [{source_type}] Harvest source: {title} [{state}]')
+            if state == 'active':
+                # We don't get full harvest soure info here. We need a custom call
+                harvest_show_url = f'{self.url}/api/3/action/harvest_source_show'
+                params = {'id': name}
+                logger.info(f'Get harvest source data {harvest_show_url} {params}')
+                response = requests.get(harvest_show_url, params=params, headers=headers)
+                if response.status_code >= 400:
+                    logger.error(f'Error [{response.status_code}] trying to get full harvest source info')
+                else:
+                    full_hs = response.json()
+                    yield full_hs['result']
+
+        # if the page is not full, is the last one
+        if count + 1 < page_size:
+            return
+
+        # get next page   
+        yield from self.list_harvest_sources(source_type=source_type, start=start + page_size, page_size=page_size)
+    
+    def get_request_headers(self, include_api_key=True):
+        headers = {'User-Agent': f'{self.user_agent}'}
+        if include_api_key:
+            headers['X-CKAN-API-Key'] = self.api_key
+        return headers
+
+    def create_harvest_source(self, data):
+        """ create a harvest source (is just a CKAN dataset/package)
+            Ensure to create the organization first.
+            params:
+                data (dict): Harvest source dict
+
+            returns:
+                created (boolean):
+                status_code (int): request status code
+                error (str): None or error
+            """
+    
+        created, status, error = self.create_organization(data=data['organization'])
+        if not created:
+            return False, status, f'Unable to create organization: {error}'
+
+        logger.info(data)
+        config = data.get('config', {})
+        if type(config) == str:
+            config = json.loads(config)
+
+        # we can also have config in extras
+        # we don't get extras 
+        extras = data.get('extras', {})
+        for extra in extras:
+            if extra['key'] == 'config':
+                logger.info(f'Config found in extras: {extra}')
+                value = json.loads(extra['value'])
+                config.update(value)
+
+        ckan_package = {
+            'name': data['name'],
+            'owner_org': data['organization']['name'],
+            'title': data['title'],
+            'url': data['url'],
+            'notes': data['notes'],
+            'source_type': data['source_type'],
+            'frequency': data['frequency'],
+            'config': json.dumps(config, indent=4)
```

Comment:
Is the JSON indent necessary? Will the extra whitespace be stripped when it's inserted into the database?

Reply:
No, it's just nice to see in the UI. I removed the indent.